tests/genrule/BUILD.bazel

```diff
@@ -0,0 +1,22 @@
+load("@aspect_bazel_lib//lib:write_source_files.bzl", "write_source_file")
+load("//helm:defs.bzl", "helm_chart")
+
+helm_chart(
+    name = "chart",
+    chart = "Chart.yaml",
+    values = "values.yaml",
+)
+
+genrule(
+    name = "genrule",
+    srcs = [":chart"],
+    outs = ["template.yaml"],
+    cmd = "$(HELM_BIN) template my-chart $(location :chart) > $@",
+    toolchains = ["@helm_toolchain//:resolved_toolchain"],
+)
+
+write_source_file(
```

Comment:
The `write_source_file(s)` macros stamp out a diff test and an executable target that will write the file from the output tree back to the source tree. The result is that a test will fail when the generated file has changed but has not been updated in source (so that the change is reviewable in a PR) and the error message contains the executable target to run to update it.

Example:

```bash
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //tests/genrule:generated_test
-----------------------------------------------------------------------------
21c21
<     app.kubernetes.io/name: genrule
---
>     app.kubernetes.io/name: genrulesdfsdf
FAIL: files "tests/genrule/template.yaml" and "tests/genrule/generated/template.yaml" differ. 

@@//tests/genrule:generated/template.yaml is out of date. To update this file, run:

    bazel run //tests/genrule:generated

```

You could use the same macro in place of the current `//:docs_diff_test` if you prefer.

Reply:
https://docs.aspect.build/rulesets/aspect_bazel_lib/docs/write_source_files/#write_source_files

Reply:
It's not important for this test and I can remove it. My thinking was that it would be nice to have snapshot tests of templates committed to source. That way when you update a helm chart, you're forced to also look at the resulting diff to the k8s manifests when reviewing the PR.

Reply:
Removed.